src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -34,31 +41,74 @@
  * name but different signatures where the signature can differ by the number of
  * input parameters or type of input parameters or both.
  * </p>
+ * <ul>
+ * <li>
+ * Property {@code modifierGroups} - control how different types of overloaded methods
+ * can be grouped by their modifiers.
+ * Type is {@code java.util.regex.Pattern[]}.
+ * Default value is {@code ""}.
+ * </li>
+ * </ul>
  * <p>
  * To configure the check:
  * </p>
  * <pre>
- * &lt;module name=&quot;OverloadMethodsDeclarationOrder&quot;/&gt;
+ * &lt;module name="OverloadMethodsDeclarationOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example of incorrect grouping of overloaded methods:
+ * </p>
+ * <pre>
+ * public void foo(int i) {} // OK
+ * public void foo(String s) {} // OK
+ * public void notFoo() {} // OK
+ * public void foo(String s, int i) {} // Violation, should be after 'foo(int i, String s)'
  * </pre>
  * <p>
  * Example of correct grouping of overloaded methods:
  * </p>
  * <pre>
- * public void foo(int i) {}
- * public void foo(String s) {}
- * public void foo(String s, int i) {}
- * public void foo(int i, String s) {}
- * public void notFoo() {}
+ * public void foo(int i) {} // OK
+ * public void foo(String s) {} // OK
+ * public void foo(String s, int i) {} // OK
+ * public void foo(int i, String s) {} // OK
+ * public void notFoo() {} // OK
  * </pre>
  * <p>
- * Example of incorrect grouping of overloaded methods:
+ * To configure the check to group static and private methods separately within their own
+ * individual groups
+ * </p>
+ * <pre>
+ * &lt;module name="OverloadMethodsDeclarationOrder"&gt;
+ *   &lt;property name="modifierGroups" value="static, private"&gt;&lt;/property&gt;
```

Comment:
Look at examples from main issue. One has a complex regexp.
> "^public .*, (protected|package), private, ^static$"

Add something like this to show it can handle more than just single, simple modifiers.